pydm/widgets/analog_indicator.py

```diff
@@ -0,0 +1,1584 @@
+        pointer_height = self._bg_size_rate * self._widget_height
+        points = [
+            QPoint(self.position, 15),
+            QPoint(self.position + 0.5*pointer_width, 0.5*pointer_height+15),
+            QPoint(self.position, pointer_height),
+            QPoint(self.position - 0.5*pointer_width, 0.5*pointer_height+15)
+        ]
+        self._painter.drawPolygon(QPolygon(points))
+
+    def draw_indicator(self):
+        """
+        Draw the selected indicator for current value.
+        """
+        self.set_position()
+        if self.position < 0 or self.position > self._widget_width:
+            return
+        self._painter.setPen(Qt.transparent)
+        self._painter.setBrush(self._indicator_color)
+        pointer_width = self._pointer_width_rate * self._widget_width
+        pointer_height = self._bg_size_rate * self._scale_height * 1.5
+        points = [
+            QPoint(self.position + 0.5 * pointer_width, 0),
+            QPoint(self.position - 0.5 * pointer_width, 0),
+            QPoint(self.position, pointer_height)
+        ]
+        self._painter.drawPolygon(QPolygon(points))
+
+    def draw_background(self):
+        """
+        Draw the background of the scale.
+        """
+        self._painter.setPen(Qt.black)
+        self._painter.setBrush(self._bg_color)
+        pointer_height = self._bg_size_rate * self._scale_height
+        bg_width = self._widget_width
+        bg_height = self._bg_size_rate * self._widget_height - 2
+        #bg_height = self._scale_height - 5
+        self._painter.drawRect(0, pointer_height, bg_width, bg_height)
+
+    def draw_minor_alarm_region(self):
+        """
+        Draw the minor alarm areas on the scale
+        """
+
+        self._painter.setPen(Qt.black)
+        self._painter.setBrush(self._minor_alarm_region_color)
+        lower_minor_alarm_width = self.calculate_position_for_value(self._lower_minor_alarm)
+
+        upper_minor_alarm_start = self.calculate_position_for_value(self._upper_minor_alarm)
+        upper_minor_alarm_width = self._widget_width - upper_minor_alarm_start
+        pointer_height = self._bg_size_rate * self._scale_height
+        minor_alarm_height = self._bg_size_rate * self._widget_height - 2
+        #minor_alarm_height = self._scale_height - 5
+
+
+        """
+        sets the pen color to alarm if the value is in the lower minor alarm region
+        """
+        if self._value < self._lower_minor_alarm and self._value > self._lower_major_alarm:
+            self._painter.setBrush(self._minor_alarm_color)
+        else:
+            self._painter.setBrush(self._minor_alarm_region_color)
+
+        self._painter.drawRect(0, pointer_height, lower_minor_alarm_width, minor_alarm_height)
+        """
+        sets the pen color to alarm if the value is in the upper minor alarm region
+        """
+        if self._value > self._upper_minor_alarm and self._value < self._upper_major_alarm:
```

Comment:
No idea how to get to this codepath, but it indicates that some care should be taken with using user input in draw functions
```
Exception occurred while running Qt Designer.
Traceback (most recent call last):
  File "/cds/home/z/zlentz/pydev/pydm/widgets/analog_indicator.py", line 284, in paintEvent
    self.draw_minor_alarm_region()
  File "/cds/home/z/zlentz/pydev/pydm/widgets/analog_indicator.py", line 208, in draw_minor_alarm_region
    if self._value > self._upper_minor_alarm and self._value < self._upper_major_alarm:
TypeError: '<' not supported between instances of 'int' and 'NoneType'
```